static/app/components/tooltip.tsx

```diff
@@ -225,7 +236,18 @@ class Tooltip extends React.Component<Props, State> {
 
     propList.containerDisplayMode = this.props.containerDisplayMode;
     return (
-      <Container {...propList} className={this.props.className} ref={ref}>
+      <Container
+        {...propList}
+        className={this.props.className}
+        ref={el => {
+          if (typeof ref === 'function') {
+            ref(el);
+          }
+          if (showOnEllipsis) {
+            this.isEllipsis = Boolean(el && isOverflown(el));
+          }
+        }}
```

Comment:
I think a more react-like pattern may be to save el in state and eval if it overflows when opening?

Reply:
I can try this again. I had issues the first time around because I ended up triggering an infinite loop with `setState` during rendering

Reply:
Done!

Reply:
Ah that makes since since it'd be updating the state after the component is mounted and causing another re-render. In that case, what if we the el was a class prop instead of `isEllipsis` - I think it might be more clear if we moved that logic where it's needed (since it isn't re-used elsewhere).